Ledger (#4)
NEAR-DevHub/treasury-docs: docs/ledger.md

---
sidebar_position: 4
sidebar_label: "Ledger"

> title: Using a ledger device

Reviewer: ```suggestion
title: Using a Ledger device
```

description: Using a ledger device to confirm your actions
---

> # About ledger devices

Reviewer: ```suggestion
# About Ledger devices
```

> A ledger device lets you confirm transactions outside your computer. If connected with a ledger device, the app you are interacting with does not have any access keys to your account, and has to ask your ledger for confirming every blockchain transaction. Your private key, which is needed to sign a transaction, is then on the ledger device. With a web wallet or mobile wallet app, that private key is stored in the wallet application, on your computer or mobile device. With a ledger, you have a dedicated device for the private keys. While there are fewer attack surfaces on a separate ledger device, compared to a web or mobile wallet, you still need to trust the ledger just as much as any other wallet software for keeping your keys safe.

Reviewer: ```suggestion
A [Ledger device](https://www.ledger.com/) lets you confirm transactions outside your computer. If connected with a ledger device, the app you are interacting with does not have any access keys to your account, and has to ask your ledger to confirm every blockchain transaction. Your private key, which is needed to sign a transaction, is then on the ledger device. With a web wallet or mobile wallet app, that private key is stored in the wallet application, on your computer or mobile device. With a ledger, you have a dedicated device for the private keys. While there are fewer attack surfaces on a separate ledger device, compared to a web or mobile wallet, you still need to trust the ledger just as much as any other wallet software for keeping your keys safe.
```

Confirming transactions with the ledger device requires a bit more effort than with a web or mobile wallet. When confirming the transaction, the ledger needs to be unlocked and the NEAR app must be open. You need to click through the transaction content and sign. Applications on BOS may time out while waiting for the transaction to confirm, and assume there's an error, while you are just in the process of clicking through the confirmation of the transaction on your ledger device.

# Prerequisites

Ledger devices have software installed to support its operations. There is an operating system, and there are apps for different blockchain. You need to make sure that the operating system and NEAR app is installed and up to date, otherwise you may get unexpected results. Consult with the documentation of your ledger device to make sure that it is updated.

Also make sure to navigate to the web4 version of your treasury, and that you are not already connected with another wallet. The direct ledger device support, described below, is only available through URLs on the form `https://<your-treasury-instance-name>.near.page`, and not through the `near.social` or `dev.near.page gateways`.

> # Using a ledger device with the treasury app

Reviewer: ```suggestion
# Using a Ledger device with the Treasury app
```
